src/codegen/doc/format.rs

```diff
@@ -54,121 +55,207 @@ fn get_language<'a>(entry: &'a str, out: &mut String) -> &'a str {
     let out = TAGS.replace_all(&out, "`$0`");
     SPACES.replace_all(&out, " ").into_owned()
 }
+
+fn find_type_by_name(symbol: &str, method_name: Option<&str>, env: &Env) -> String {
+    let symbols = env.symbols.borrow();
+    if let Some(method) = method_name {
+        if let Some((obj_info, fn_info)) = env.analysis.find_object_by_function(
+            env,
+            |o| o.full_name == symbol,
+            |f| f.name == method,
+        ) {
+            let sym = symbols.by_tid(obj_info.type_id).unwrap(); // we are sure the object exists
+            let (type_name, visible_type_name) = obj_info.generate_doc_link_info(fn_info);
+            let name = sym.full_rust_name().replace(&obj_info.name, &type_name);
+
+            fn_info.doc_link(Some(&name), Some(&visible_type_name))
+        } else if let Some((record_info, fn_info)) = env.analysis.find_record_by_function(
+            env,
+            |r| r.type_(&env.library).c_type == symbol,
+            |f| f.name == method,
+        ) {
+            let sym_name = symbols
+                .by_tid(record_info.type_id)
+                .unwrap()
+                .full_rust_name(); // we are sure the object exists
+            fn_info.doc_link(Some(&sym_name), None)
+        } else {
+            format!("`{}::{}`", symbol, method)
+        }
+    } else {
+        find_type(symbol, env)
+    }
+}
+
+fn find_constant_or_variant(symbol: &str, env: &Env) -> String {
+    let symbols = env.symbols.borrow();
+    if let Some(const_info) = env
+        .analysis
+        .constants
+        .iter()
+        .find(|c| c.glib_name == symbol)
+    {
+        // for whatever reason constants are not part of the symbols list
+        format!("[`{name}`][crate::{name}]", name = const_info.name)
+    } else if let Some((flag_info, member_info)) = env.analysis.flags.iter().find_map(|f| {
+        f.type_(&env.library)
+            .members
+            .iter()
+            .find(|m| m.c_identifier == symbol)
+            .map(|m| (f, m))
+    }) {
+        let sym = symbols.by_tid(flag_info.type_id).unwrap();
+        format!(
+            "[`{flag_name}::{member_name}`][crate::{parent}::{member_name}]",
+            member_name = nameutil::bitfield_member_name(&member_info.name),
+            flag_name = flag_info.name,
+            parent = sym.full_rust_name()
+        )
+    } else if let Some((enum_info, member_info)) = env.analysis.enumerations.iter().find_map(|e| {
+        e.type_(&env.library)
+            .members
+            .iter()
+            .find(|m| m.c_identifier == symbol)
+            .map(|m| (e, m))
+    }) {
+        let sym = symbols.by_tid(enum_info.type_id).unwrap();
+        format!(
+            "[`{enum_name}::{member}`][crate::{parent}::{member}]",
+            enum_name = enum_info.name,
+            member = nameutil::enum_member_name(&member_info.name),
+            parent = sym.full_rust_name()
+        )
```

Comment:
Same here, `enum_info.name` is always equivalent to `sym.full_rust_name()`, do we need this?

Reply:
Needed, not really :) I think i did that when the enum members was breaking the symbols list